client/db/bolt/upgrades_test.go

```diff
@@ -150,6 +128,39 @@ func verifyV2Upgrade(t *testing.T, db *bbolt.DB) {
 	}
 }
 
+func verifyV3Upgrade(t *testing.T, db *bbolt.DB) {
+	verifyMatches := map[string]bool{
+		"090f54bc6b13652d4e4c23ec94b4f70751c61db043ae732c014d0f5cf4ccccb7": false, // status < MatchComplete, not revoked, not refunded
+		"5ee45ebb29edf84ab2a7f35aef619b20c3e2cd26017a5e281e7cc1bfde0ed435": true,  // status < MatchComplete, revoked at NewlyMatched
+		"84455dae423b09f149396f09b6063380eeb312e1b67051a695e78cd5c985cb49": false, // status == MakerSwapCast, side Maker, revoked, requires refund
+		"59c054d537fbd98f51621671a84d25b55350d0e33a6b3c2c4abd874054f8a438": true,  // status == TakerSwapCast, side Maker, revoked, refunded
+		"f153a3b8d5824c3e237911cd1b108dfb2e04555f6c5899681d182216caa328fc": false, // status == MatchComplete, no RedeemSig
+		"3996ffea2a3db99262b954cf6dcf883f562baf273d767b7cf91b6aacddd4206f": true,  // status == MatchComplete, RedeemSig set
+	}
```

Comment:
I wrote [a tool](https://github.com/itswisdomagain/dcrdex/blob/00d492e9c8a5dd98b6720e102d497a4684596800/client/db/bolt/testdata/v2.db.gen.go) to generate these matches using the previous db version and log the ids and expected retirement status for each generated match.

Can verify by running this test with the `v2.db.gz` file generated by that tool and pasting the log output here.

Reply:
I'd be good with checking that in if you give it a build tag like `// +build ignore` so it doesn't break anything.

Reply:
Looks like that file will have to go in before this PR because it cannot be executed if it goes in with this PR. If it goes in a preceding commit, it can be executed by checking out that commit. Separate PR or nah?

Reply:
I had meant to respond to this with "preceding commit" for the gen tool.

Although I keep going back to @JoeGruffins' active orders bucket upgrade (`v4Upgrade` and the `moveActiveOrders` helper) as a model solution for the matches too.  I don't think we considered this when this work was initiated, but it not only solves the problem of knowing which matches in the DB are active or retired, but it provides a major performance boost by not having to iterate over ancient matches.

Only thing I'm not sure about is if there's an active order that has some retired matches, would it become a burden to have to check two match buckets for such an order's matches (we display all matches for active orders including the inactive ones).

I will poke around to see.  This Retired flag does seem to work, but if it would be obsoleted with a separate matches bucket, I'd hate to go ahead with the DB upgrade, esp. because we've seen OOM issues with these upgrades on large DBs and updating every entry of the matches bucket is likely to be the worst case scenario.

I will test some things with the current PR and look at these ideas.  No need to do anything yet.  Let's just prioritize #1210 for now.